### ISSUE TITLE
Become error, where work fine before, in locked package.json

### ISSUE THREAD

**Issue by VitalyName** [protobufjs/protobuf.js#1413]
protobuf.js version: 6.8.8
nodejs version: 6.11.0

I have the problem now, but i don't have problem 1 year before. This is strange, because dependency version is locked, npm-shrinkwrap.json is used

After run command
```js
pbjs -t static-module -w commonjs -o ../../../target/classes/myproto.js ../proto/*.proto; pbts -o ../../../target/classes/myproto.d.ts ../../../target/classes/myproto.js
```
I can see this error:
```js
node_modules/protobufjs/cli/node_modules/mkdirp/lib/opts-arg.js:7
    opts = { mode: 0o777, fs, ...opts }
                              ^^^
SyntaxError: Unexpected token ...
```
Probably "...opts" is syntax, which nodejs6.11.0 doesn't support. But why it works before?
What can I do for fix it in my project, without update nodejs version?

**Comment by VitalyName** [comment]
I try different node version. With them I don't see same syntax error, but then I have issue with missing interfaces in typescript file, like this [https://github.com/protobufjs/protobuf.js/issues/1222](https://github.com/protobufjs/protobuf.js/issues/1222)

**Comment by VitalyName** [comment]
I found the reason for this issue. list of dependencies for protobuf.js doesn't locked for minor version. So for me it become not working now because mkdirp library was updated, and protobuf.js take new mkdirp. It new mkdirp contain new js syntax which nodejs6.11.0 doesen't support. 

it is very sad when my project was working well before, now it become broken, although I did not make any changes to my project. This would not happen if brotobuf.js dependecies was locked for path version instead of minor.

**Comment by VitalyName** [comment]
I understood the cause of the problem. This is dependency protobufjs. In package json of protobufjs 6.8.8 is devDependency    "jsdoc": ^3.5.5. 

Year ago, the most recent version was jsdoc@3.5.5. Where I build my project, prodobuf.js/cli install jsdoc@3.5.5. jsdoc@3.5.5 depends on mkdirp@0.5.1. mkdirp@0.5.1 work with node6.11.0 fine.

Year ago, the most recent version was jsdoc@3.6.4 and where I build my project, prodobuf.js/cli install jsdoc@3.6.4. It depends on mkdirp@1.0.4, which contain new js syntax an it doesn't work with node 6.11.0.

I don't understand why protobuf CLI install library only when I run pbjs commands. Why it not installed where i build my project. 

I can not affect to jsdoc version in my package-lock.json because jsdoc is in  devDependency. I am very disapponted, my project which building well before, now cannot be build, although I did not make any changes to it.

Maybe there are ways to resolve this? 

I updated nodejs, but i got new issue related to jsdoc version [https://github.com/protobufjs/protobuf.js/issues/1414](https://github.com/protobufjs/protobuf.js/issues/1414)
